typegen/strats/inline.py

```diff
@@ -0,0 +1,257 @@
+                params[TraceData.LINENO] == arg.lineno,
+                params[TraceData.VARNAME] == arg.arg,
+            ]
+            arg_hints = params[functools.reduce(operator.and_, arg_hint_mask)]
+            assert (
+                arg_hints.shape[0] <= 1
+            ), f"Found multiple hints for the parameter type: {arg_hints}"
+
+            # no type hint, skip
+            if arg_hints.shape[0] == 0:
+                logger.debug(f"No hint found for '{arg.arg}'")
+                continue
+
+            arg_hint = arg_hints[TraceData.VARTYPE].values[0]
+            logger.debug(f"Applying hint '{arg_hint}' to '{arg.arg}'")
+            arg.annotation = ast.Name(arg_hint)
+
+    def _add_return_hint(self, fdef: ast.FunctionDef) -> None:
+        # disambiguate methods from functions
+        clazz_name = self._find_class(fdef)
+
+        if clazz_name is not None:
+            clazz_mask = self.df[TraceData.CLASS].map(lambda t: t.__name__ if t else None) == clazz_name
+        else:
+            clazz_mask = self.df[TraceData.CLASS].isnull()
+
+        rettype_masks = [
+            self.df[TraceData.CATEGORY] == TraceDataCategory.FUNCTION_RETURN,
+            clazz_mask,
+            self.df[TraceData.VARNAME] == fdef.name,
+            self.df[TraceData.LINENO] == 0,  # return type, always stored at line 0
+        ]
+        rettypes = self.df[functools.reduce(operator.and_, rettype_masks)]
+
+        assert (
+            rettypes.shape[0] <= 1
+        ), f"Found multiple hints for the return type:\n{rettypes[TraceData.VARTYPE].values}"
+
+        # no type hint, skip
+        if rettypes.shape[0] == 0:
+            logger.debug(f"No hint found for return for '{fdef.name}'")
+
+        if rettypes.shape[0] == 1:
+            ret_hint = rettypes[TraceData.VARTYPE].values[0]
+            logger.debug(f"Applying return type hint '{ret_hint}' to '{fdef.name}'")
+            fdef.returns = ast.Name(ret_hint)
+
+    def _find_class(self, node: ast.AST) -> str | None:
+        class_name = None
+        while hasattr(node, "parent"):
+            node = node.parent  # type: ignore
+            if isinstance(node, ast.ClassDef):
+                class_name = node.name
+                break
+
+        return class_name
+
+    def _find_targets(self, node: ast.Assign | ast.AnnAssign | ast.AugAssign):
+        if isinstance(node, ast.AnnAssign | ast.AugAssign):
+            target = node.target
+            return self._extract_target_names_with_nodes(target)
+        else:
+            targets = node.targets
+            names = list()
+            for target in targets:
+                names += self._extract_target_names_with_nodes(target)
+            return names
+
+    def visit_AugAssign(self, node: ast.AugAssign) -> ast.AST | list[ast.AST]:
+        assigns = self._visit_assigns(node)
+        # Overwrite with actual assignment
+        assert isinstance(assigns, list)
+        assigns[-1] = node
```

Comment:
Should already be covered in line 218. Remove this line.

Reply:
Done